### DISCUSSION TITLE
Have you tried to implement pdf.js using asp.net core?

### DISCUSSION THREAD

**Discussion by imbenjogadia** [mozilla/pdf.js#17452] [Q&A]
Hi everyone, I hope you are doing well. I need your help, I am trying to implement the pdf.js viewer using asp.net core, have anyone tried it before? I am getting an error on the **/lib/pdfjs/web/locale/en-US/viewer.ftl**. I am using the pdf.js version 4.0.269

![image](https://github.com/mozilla/pdf.js/assets/42939920/94bf9de3-a9c3-415e-b902-cc4be2f29687)

Thank you.

**Comment by fmmich** [comment]
By default, Kestrel doesn't recognize ".ftl" as a valid web extension.

Add this to your Program.cs or wherever you build the web app, where "app" is WebApplication:
```
var contentProvider = new FileExtensionContentTypeProvider();
contentProvider.Mappings[".ftl"] = "application/octet-stream";
app.UseStaticFiles(new StaticFileOptions
{
    ContentTypeProvider = contentProvider
});
```

Reference: https://learn.microsoft.com/en-us/aspnet/core/fundamentals/static-files?view=aspnetcore-8.0#fileextensioncontenttypeprovider
